packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionHistory.tsx

```diff
@@ -0,0 +1,144 @@
+import { Tab } from '@headlessui/react'
+import { Dispatch, Fragment, SetStateAction, useMemo } from 'react'
+import { CompleteDepositData } from '../../hooks/useDeposits'
+import { useNetworksAndSigners } from '../../hooks/useNetworksAndSigners'
+import { CompleteWithdrawalData } from '../../hooks/useWithdrawals'
+import { useAppState } from '../../state'
+import { getNetworkLogo, getNetworkName } from '../../util/networks'
+import {
+  PageParams,
+  TransactionsTable
+} from './TransactionsTable/TransactionsTable'
+import { PendingTransactions } from './PendingTransactions'
+import { FailedTransactionsWarning } from './FailedTransactionsWarning'
+import { isFailed, isPending } from '../../state/app/utils'
+
+export const TransactionHistory = ({
+  depositsPageParams,
+  withdrawalsPageParams,
+  depositsData,
+  depositsLoading,
+  depositsError,
+  withdrawalsData,
+  withdrawalsLoading,
+  withdrawalsError,
+  setDepositsPageParams,
+  setWithdrawalsPageParams
+}: {
+  depositsPageParams: PageParams
+  withdrawalsPageParams: PageParams
+  depositsData: CompleteDepositData
+  withdrawalsData: CompleteWithdrawalData
+  depositsLoading: boolean
+  depositsError: boolean
+  withdrawalsLoading: boolean
+  withdrawalsError: boolean
+  setDepositsPageParams: Dispatch<SetStateAction<PageParams>>
+  setWithdrawalsPageParams: Dispatch<SetStateAction<PageParams>>
+}) => {
+  const { l1, l2 } = useNetworksAndSigners()
+
+  const {
+    app: { mergedTransactions }
+  } = useAppState()
+
+  const pendingTransactions = useMemo(() => {
+    return mergedTransactions?.filter(tx => isPending(tx))
+  }, [mergedTransactions])
+
+  const failedTransactions = useMemo(() => {
+    return mergedTransactions?.filter(tx => isFailed(tx))
+  }, [mergedTransactions])
+
+  const localCachedTransactionMap = useMemo(() => {
```

Comment:
Hm, this is a bit confusing for me. It has nothing to do to transactions cached in the local storage?